mvn-extension/src/main/java/org/arquillian/smart/testing/mvn/ext/PreBuildTestOptimizer.java

```diff
@@ -75,8 +75,8 @@ private void configureExtension(MavenSession session, Configuration configuratio
         final MavenProjectConfigurator mavenProjectConfigurator = new MavenProjectConfigurator(configuration);
         session.getAllProjects().forEach(mavenProject -> {
             final Model model = mavenProject.getModel();
-            mavenProjectConfigurator.addRequiredDependencies(model);
             mavenProjectConfigurator.configureTestRunner(model);
+            mavenProjectConfigurator.addRequiredDependencies(model);
```

Comment:
I think that order is important I am not 100% sure but for example `configureTestRunner` checks the surefire version is the minimum required one. So probably it is better to first detect these things and then add the dependencies and not the other way  around.